In [ ]:
!pip install evaluate

   ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 84.0/84.0 kB 9.2 MB/s eta 0:00:00
   ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 484.9/484.9 kB 42.4 MB/s eta 0:00:00
   ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 116.3/116.3 kB 14.5 MB/s eta 0:00:00
   ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 143.5/143.5 kB 16.3 MB/s eta 0:00:00
   ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 194.8/194.8 kB 22.3 MB/s eta 0:00:00


In [ ]:
!pip install rouge_score

  Preparing metadata (setup.py) ... done
  Created wheel for rouge_score: filename=rouge_score-0.1.2-py3-none-any.whl size=24935 sha256=fc1a47bf0c6fd0b872596379c31eb40a0735d519de25f5d5b00a8d52b865e416
  Stored in directory: /root/.cache/pip/wheels/1e/19/43/8a442dc83660ca25e163e1bd1f89919284ab0d0c1475475148
Successfully built rouge_score


In [ ]:
# prompt: import an xlsx file

import pandas as pd

# Replace 'your_file.xlsx' with the actual path to your Excel file
# If the file is in your current working directory, you can just use the filename
file_path = '/content/df_fr_fri_test.xlsx'

try:
  dataset = pd.read_excel(file_path)
  print(dataset.head()) # Print the first few rows to verify
except FileNotFoundError:
  print(f"Error: File '{file_path}' not found.")
except Exception as e:
  print(f"An error occurred: {e}")


                                                  fr  \
0         Nombre de participants : non déterminant\n   
1  Le message est clair : ces organisations et pe...   
2  Toutes les informations utiles sur la sécurité...   
3  Ce projet vise à sélectionner, sous-titrer et ...   
4  Au lieu de chercher à ouvrir le champ des poss...   

                                                 fri  \
0        Nombre de participant.e.s : non déterminant   
1  Le message est clair : ces organisations et pe...   
2  Toutes les informations utiles sur la sécurité...   
3  Ce projet vise à sélectionner, sous-titrer et ...   
4  Au lieu de chercher à ouvrir le champ des poss...   

                              fr_inclusive_predicted  
0         Nombre de participants: non déterminant \n  
1  Le message est clair: ces organisations et per...  
2  Toutes les informations utiles sur la sécurité...  
3  Ce projet vise à sélectionner, sous-titrer et ...  
4  Au lieu de chercher à ouvrir le champ des poss..

In [ ]:
# Prepare references and predictions
# The lambda function iterates over the entire thing and put it into a list format
# The metrics expect the references to be a list of lists,
# even if there’s only one reference sentence per prediction.
# BLEU and ROUGE expect the predictions to be a simple list (not a list of lists).
references = dataset['fri'].apply(lambda x: [x]).tolist()
predictions = dataset['fr_inclusive_predicted'].tolist()

In [ ]:
import evaluate

# Load evaluation metrics
bleu = evaluate.load("bleu")
rouge = evaluate.load("rouge")

# Compute BLEU
bleu_score = bleu.compute(predictions=predictions, references=references)

# Compute ROUGE
rouge_score = rouge.compute(predictions=predictions, references=references)

print("BLEU:", bleu_score)
print("ROUGE:", rouge_score)

/usr/local/lib/python3.11/dist-packages/huggingface_hub/utils/_auth.py:94: UserWarning: 
The secret `HF_TOKEN` does not exist in your Colab secrets.
To authenticate with the Hugging Face Hub, create a token in your settings tab (https://huggingface.co/settings/tokens), set it as secret in your Google Colab and restart your session.
You will be able to reuse this secret in all of your notebooks.
Please note that authentication is recommended but still optional to access public models or datasets.
  warnings.warn(


BLEU: {'bleu': 0.7413216079690788, 'precisions': [0.9304235115758288, 0.881054655906828, 0.839577234158686, 0.8042233886044315], 'brevity_penalty': 0.8594615238886983, 'length_ratio': 0.8684707774260013, 'translation_length': 172817, 'reference_length': 198990}
ROUGE: {'rouge1': 0.829918159829493, 'rouge2': 0.7232193828945028, 'rougeL': 0.8288402933263805, 'rougeLsum': 0.8287787374751732}


Results:

BLEU: {'bleu': 0.7413216079690788, 'precisions': [0.9304235115758288, 0.881054655906828, 0.839577234158686, 0.8042233886044315], 'brevity_penalty': 0.8594615238886983, 'length_ratio': 0.8684707774260013, 'translation_length': 172817, 'reference_length': 198990}
ROUGE: {'rouge1': 0.8298654517303206, 'rouge2': 0.7229960603902744, 'rougeL': 0.8287127804099808, 'rougeLsum': 0.8289103128223406}

In [ ]:
!pip install unbabel-comet

     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 55.5/55.5 kB 5.5 MB/s eta 0:00:00
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 51.8/51.8 kB 5.7 MB/s eta 0:00:00
   ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 96.2/96.2 kB 10.6 MB/s eta 0:00:00
   ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 101.4/101.4 kB 12.0 MB/s eta 0:00:00
   ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 819.3/819.3 kB 47.5 MB/s eta 0:00:00
   ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 104.1/104.1 kB 10.8 MB/s eta 0:00:00
   ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 363.4/363.4 MB 4.3 MB/s eta 0:00:00
   ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 13.8/13.8 MB 101.3 MB/s eta 0:00:00
   ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 24.6/24.6 MB 86.0 MB/s eta 0:00:00
   ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 883.7/883.7 kB 57.0 MB/s eta 0:00:00
   ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 664.8/664.8 MB 3.1 MB/s eta 0:00:00
   ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 211.5/211.5 MB 5.8 MB/s eta 0:00:00
   ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━

In [ ]:
# Check for any NaN values or unexpected types
print("Any NaNs in 'fr' column:", dataset['fr'].isna().sum())
print("Any NaNs in 'fr_inclusive_predicted' column:", dataset['fr_inclusive_predicted'].isna().sum())
print("Any NaNs in 'fri' column:", dataset['fri'].isna().sum())

# Check the data types in the columns
print("Data types of 'fr' column:", dataset['fr'].dtype)
print("Data types of 'fr_inclusive_predicted' column:", dataset['fr_inclusive_predicted'].dtype)
print("Data types of 'fri' column:", dataset['fri'].dtype)


Any NaNs in 'fr' column: 6
Any NaNs in 'fr_inclusive_predicted' column: 6
Any NaNs in 'fri' column: 6
Data types of 'fr' column: object
Data types of 'fr_inclusive_predicted' column: object
Data types of 'fri' column: object


In [ ]:
# Check for any rows where 'fr' or 'fr_inclusive_predicted' or 'fri' are floats or other unexpected types
print(dataset[dataset['fr'].apply(lambda x: isinstance(x, float))].head())
print(dataset[dataset['fr_inclusive_predicted'].apply(lambda x: isinstance(x, float))].head())
print(dataset[dataset['fri'].apply(lambda x: isinstance(x, float))].head())


       fr  fri fr_inclusive_predicted
293   NaN  NaN                    NaN
733   NaN  NaN                    NaN
2415  NaN  NaN                    NaN
3259  NaN  NaN                    NaN
4384  NaN  NaN                    NaN
       fr  fri fr_inclusive_predicted
293   NaN  NaN                    NaN
733   NaN  NaN                    NaN
2415  NaN  NaN                    NaN
3259  NaN  NaN                    NaN
4384  NaN  NaN                    NaN
       fr  fri fr_inclusive_predicted
293   NaN  NaN                    NaN
733   NaN  NaN                    NaN
2415  NaN  NaN                    NaN
3259  NaN  NaN                    NaN
4384  NaN  NaN                    NaN


In [ ]:
print(f"Rows before cleaning: {dataset.shape[0]}")

# Drop rows where any of the relevant columns have NaN values
dataset_clean = dataset.dropna(subset=['fr', 'fr_inclusive_predicted', 'fri'])

# Verify that the NaNs were removed
print(f"Rows after cleaning: {dataset_clean.shape[0]}")

Rows before cleaning: 5900
Rows after cleaning: 5894


In [ ]:
from comet import download_model, load_from_checkpoint

# Load a pre-trained COMET model
comet_model_path = download_model("Unbabel/wmt20-comet-da")
comet_model = load_from_checkpoint(comet_model_path)

# Prepare data
'''
COMET requires a source, reference, and hypothesis (prediction)
It assesses fluency sound of the translations

Sources (src): original French sentences
Predictions (mt):  inclusive French translations produced by the model
References (ref): standard French translations
'''
# Prepare data
comet_data = [{"src": src, "mt": pred, "ref": ref[0]} for src, pred, ref in zip(dataset_clean['fr'], dataset_clean['fr_inclusive_predicted'], dataset_clean['fri'].apply(lambda x: [x]))]

# Compute COMET score
comet_score = comet_model.predict(comet_data)

print("COMET Score:", comet_score)


Fetching 5 files:   0%|          | 0/5 [00:00<?, ?it/s]

.gitattributes:   0%|          | 0.00/1.48k [00:00<?, ?B/s]

hparams.yaml:   0%|          | 0.00/437 [00:00<?, ?B/s]

README.md:   0%|          | 0.00/3.58k [00:00<?, ?B/s]

LICENSE:   0%|          | 0.00/9.69k [00:00<?, ?B/s]

model.ckpt:   0%|          | 0.00/2.33G [00:00<?, ?B/s]

INFO:pytorch_lightning.utilities.migration.utils:Lightning automatically upgraded your loaded checkpoint from v1.3.5 to v2.5.0.post0. To apply the upgrade to your files permanently, run `python -m pytorch_lightning.utilities.upgrade_checkpoint ../root/.cache/huggingface/hub/models--Unbabel--wmt20-comet-da/snapshots/87819f4d6d4f17e0d1752cc9e0ccfa2064997219/checkpoints/model.ckpt`


tokenizer_config.json:   0%|          | 0.00/25.0 [00:00<?, ?B/s]

sentencepiece.bpe.model:   0%|          | 0.00/5.07M [00:00<?, ?B/s]

tokenizer.json:   0%|          | 0.00/9.10M [00:00<?, ?B/s]

config.json:   0%|          | 0.00/616 [00:00<?, ?B/s]

/usr/local/lib/python3.11/dist-packages/pytorch_lightning/core/saving.py:195: Found keys that are not in the model state dict but in the checkpoint: ['encoder.model.embeddings.position_ids']
INFO:pytorch_lightning.utilities.rank_zero:GPU available: True (cuda), used: True
INFO:pytorch_lightning.utilities.rank_zero:TPU available: False, using: 0 TPU cores
INFO:pytorch_lightning.utilities.rank_zero:HPU available: False, using: 0 HPUs
INFO:pytorch_lightning.accelerators.cuda:LOCAL_RANK: 0 - CUDA_VISIBLE_DEVICES: [0]
Predicting DataLoader 0: 100%|██████████| 369/369 [02:59<00:00,  2.05it/s]


COMET Score: Prediction([('scores', [1.2316406965255737, 1.239678144454956, 0.9668121933937073, 0.6307732462882996, 0.7228959798812866, 1.228714942932129, 0.6013273000717163, 0.9173915982246399, 0.8534250259399414, 0.5987737774848938, 1.1374115943908691, 1.1422817707061768, -0.17410986125469208, 0.9198635220527649, 1.0499802827835083, 1.2207804918289185, 1.0609275102615356, 1.202713966369629, 1.1178512573242188, 1.3096532821655273, 1.1893353462219238, 0.6868262887001038, 1.1316430568695068, 1.2252988815307617, 0.07272830605506897, 0.9701414108276367, 1.2662665843963623, 0.8489475846290588, 0.04003848135471344, 0.05878197401762009, 1.0324406623840332, 1.2405842542648315, 0.9465309381484985, 1.2693506479263306, 1.147087812423706, 1.170215368270874, 1.1253682374954224, 1.2106480598449707, 1.2040005922317505, 1.0150384902954102, 1.294044017791748, 1.161781907081604, 1.040992021560669, 1.1218030452728271, 1.0729671716690063, -0.01922905072569847, 0.29078346490859985, 1.24237060546875, 1.206

COMET results: ('system_score', 0.8718611426701409)